src/commands/integrations/flows/test.ts

```diff
@@ -116,145 +136,210 @@ export default class TestFlowCommand extends PrismaticBaseCommand {
       },
     } = await this.parse(TestFlowCommand);
 
-    // Load custom trigger payload if provided.
     let triggerPayload = "";
+    let payloadHeaders: Record<string, string> = {};
+    let effectiveContentType = contentType;
+    let replayablePayload: ReplayablePayload | null = null;
+
     if (payloadFilePath) {
-      if (await exists(payloadFilePath)) {
-        triggerPayload = await fs.readFile(payloadFilePath, { encoding: "utf-8" });
+      if (!(await exists(payloadFilePath))) {
+        handleError({
+          message: `No file found at ${payloadFilePath}. Please check the --trigger-payload-file (-p) parameter.`,
+          throwError: true,
+        });
+      }
 
-        // Validate JSON files specifically. CSV and XML files fail more gracefully if malformed
+      const fileContent = await fs.readFile(payloadFilePath, { encoding: "utf-8" });
+      replayablePayload = parseReplayablePayload(fileContent);
+
+      if (replayablePayload) {
+        // If the given payload file is from listening mode replay, we will parse it for
+        // the body, content type, and headers, and merge that in with the given flags accordingly.
+        triggerPayload =
+          typeof replayablePayload.payload === "string"
+            ? replayablePayload.payload
+            : JSON.stringify(replayablePayload.payload);
+
+        const hasExplicitContentType =
+          this.argv.includes("-c") ||
+          this.argv.some(
+            (arg) => arg.startsWith("-c=") || arg.startsWith("--payload-content-type"),
+          );
+        if (!hasExplicitContentType && replayablePayload.contentType) {
+          effectiveContentType = replayablePayload.contentType;
+        }
+
+        if (typeof replayablePayload.headers === "string") {
+          try {
+            payloadHeaders = JSON.parse(replayablePayload.headers);
+          } catch {
+            throw new Error(
+              `There was an error parsing the headers listed in the payload file: ${payloadFilePath}`,
+            );
+          }
+        } else if (
+          typeof replayablePayload.headers === "object" &&
+          replayablePayload.headers !== null
+        ) {
+          payloadHeaders = replayablePayload.headers;
+        }
```

Comment:
As we think more about testing this seems like an area that would be nice to potentially break up and test given the amount of conditionality here. I'm not sure what the right organization structure is quite yet but theres a fair amount of density here that might make it tricky to evolve.

Reply:
No action needed now, just something to keep in mind as we pursue testing.

Reply:
No I totally agree with you. It'd probably be better to break this up into more functions or something